packages/lesswrong/lib/collections/posts/helpers.ts

```diff
@@ -200,3 +201,62 @@ export const postGetKarma = (post: PostsBase|DbPost): number => {
 export const postCanEditHideCommentKarma = (user: UsersCurrent|DbUser|null, post?: PostsBase|DbPost|null): boolean => {
   return !!(user?.showHideKarmaOption && (!post || !postGetCommentCount(post)))
 }
+
+/**
+ * Returns the event datetimes in a user-friendly format,
+ * ex: Mon, Jan 3 at 4:30 - 5:30 PM
+ * 
+ * @param {(PostsBase|DbPost)} post - The event to be checked.
+ * @returns {string} The formatted event datetimes.
+ */
+export const prettyEventDateTimes = (post: PostsBase|DbPost): string => {
+  // when no start time, just show "TBD"
+  let eventTime = 'TBD';
+
+  if (post.startTime) {
+    let start = moment(post.startTime)
+    let end = post.endTime && moment(post.endTime)
+    // if we have event times in the local timezone, use those instead
+    const useLocalTimes = post.localStartTime && (!post.endTime || post.localEndTime)
+    if (useLocalTimes) {
+      start = moment(post.localStartTime).utc()
+      end = post.localEndTime && moment(post.localEndTime).utc()
+    }
+    
+    // hide the year if it's reasonable to assume it
+    const now = moment()
+    const sixMonthsFromNow = moment().add(6, 'months')
+    const startYear = (now.isSame(start, 'year') || start.isBefore(sixMonthsFromNow)) ? '' : `, ${start.format('YYYY')}`
+    
+    const startDate = start.format('ddd, MMM D')
+    const startTime = start.format('h:mm')
+    let startAmPm = ` ${start.format('A')}`
+    const timezone = useLocalTimes ? '' : ` ${start.format('[UTC]ZZ')}`
```

Comment:
What is the [UTC] doing there? I'd guess it's displayed literally because I couldn't see it in the moment docs, but I also don't see the brackets in the example output.

Reply:
Bigger (potentially dumber) question: if this is running on the server, and we don't call moment.utc for non-local time, does that mean this will always be UTC-0000 on production?

Reply:
Yup the brackets are for [escaping characters](https://momentjs.com/docs/#/displaying/format/). So it would look like `UTC-0500`, which is not great but at least conveys the info.

And def not a dumb question - yeah, it actually seems to always show up as `UTC+0000`, based on the emails I've seen. But either way this is not pretty and I wanted to be sure that we give the user the correct info.

I think part of the fix is for us to use the local times more. We actually didn't have the Google Maps API set up properly, so we don't have any event local times from before this week when I fixed that on prod. Now that that is working (hopefully), most events should be using local times.

The other thing we can do is, I think Jim added a feature where we are collecting users' timezones? I think ideally we would convert the start/end times to the recipient's timezone before emailing them. I'll add this to my list.